### PR TITLE
Add EVM benchmarking scripts

### PR DESCRIPTION
**Pull request by hugo-dc**
According to this issue: https://github.com/ewasm/benchmarking/issues/22
This PR adds the EVM benchmarking scripts and separates the VM's in different Docker files.

@chfast @cdetrio please let me know if you have any comment

### CODE REVIEW

**Comment by chfast** [comment]
Please check the LGTM warnings.

**Comment by chfast** [comment]
Hey @hugo-dc, the https://github.com/ethereum/go-ethereum/pull/20330 has just been merged, so we can use go-ethereum/master now, and any future release of it.

Changes are looking good, I will review the python code in detail later. Still pay attention to LGTM warnings.

File: evm/scripts/merge.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    with open(RESULT_FILE, 'w', newline='') as bench_result_file:
+        writer = csv.DictWriter(bench_result_file, fieldnames=fieldnames)
+        writer.writeheader()
+
+    for i in range(0, len(evm_results[0])):
+        for e in range(0, len(EVMS)):
+            if i == 0:  # skip header
+                continue
+            merged_data.append(evm_results[e][i])
```
Comment (jwasinger) [inline evm/scripts/merge.py]: newline needed

File: evm/scripts/merge.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+import csv
+import os
+
+RESULT_CSV_OUTPUT_PATH = "/evmraceresults/"
+EVMS = ["evmone", "parity", "geth", "cita-vm"]
+RESULT_FILE = os.path.join(RESULT_CSV_OUTPUT_PATH + "evm_benchmarks.csv")
+
+def main():
+    fieldnames = ['engine', 'test_name', 'total_time', 'gas_used']
```
Comment (jwasinger) [inline evm/scripts/merge.py]: gas used  for evmone doesn't use same format as other clients.  e.g. `1.5K` for evmone vs `1500` for other clients.

**Pull request merged**